include/lbann/comm.hpp

```diff
@@ -832,6 +870,27 @@ class lbann_comm {
     return is_world_rank_on_node(world_rank);
   }
 
+  #ifdef LBANN_HAS_CUDA
```

Comment:
These functions allow the cuDNN manager to send the GPU list and CUDA stream list to the LBANN communicator. This is admittedly a hacky, short-term solution. In a future refactor, I think it would be worthwhile to eliminate the cuDNN manager and have the communicator manage the various GPU objects.